sdk/servicebus/Azure.Messaging.ServiceBus/tests/Receiver/ReceiverLiveTests.cs

```diff
@@ -1265,6 +1282,131 @@ await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitionin
             }
         }
 
+        [Test]
+        public async Task PurgeMessages()
+        {
+            await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: false))
+            {
+                await using var client = new ServiceBusClient(TestEnvironment.ServiceBusConnectionString);
+
+                var messageCount = ServiceBusReceiver.MaxDeleteMessageCount * 3;
+                await SendMessagesAsync(client, scope.QueueName, messageCount);
+
+                var receiver = client.CreateReceiver(scope.QueueName);
+                var numMessagesDeleted = await receiver.PurgeMessagesAsync();
+
+                Assert.GreaterOrEqual(numMessagesDeleted, messageCount);
```

Comment:
Why wouldn't we be able to assert on equality here?

Reply:
I was going to make a comment about the service implementation being quite buggy, but when I traced it back there's a bug in the change that I made to the "SendMessages" method in the base.   I think with that change, this can be straight equality.   Updated.